src/Core/src/Platform/iOS/MauiLabel.cs

```diff
@@ -83,7 +83,17 @@ public override void InvalidateIntrinsicContentSize()
 			}
 		}
 
-		public override SizeF SizeThatFits(SizeF size) => AddInsets(base.SizeThatFits(size));
+		public override SizeF SizeThatFits(SizeF size)
+		{
+			var requestedSize = base.SizeThatFits(size);
+
+			// Let's be sure the label.Width is not larger than the container
+			return AddInsets(new Size()
+			{
+				Width = nfloat.Min(requestedSize.Width, size.Width),
+				Height = requestedSize.Height
```

Comment:
This does things for width only... What happens on a wrapped label with multiple lines? Does it stay inside? Clip?

Reply:
Currently, the height will go past the label as well. I added the height cutoff just now as well. Inside a vertical stacklayout, it will not cut because the height constraint of the vsl is infinite when it gets passed into the viewhandlerextension, but it does cut it off in the grid since the height constraint is passed into there.

<img height="400" alt="WithHeightCutAndNoTruncation" src="https://user-images.githubusercontent.com/50846373/236555445-7c7d025b-4ae1-47c7-9366-fb5ad18f24e0.png">
<img height="400" alt="WithHeightCutInGrid" src="https://user-images.githubusercontent.com/50846373/236555455-cc2c4658-143a-4fe9-b750-c43e656c3af5.png">

Reply:
This seems to be similar behavior to how android handles this

<img height="408" alt="image" src="https://user-images.githubusercontent.com/50846373/236556949-95c61245-e43b-4594-a3a8-a073c9050304.png">

<img height="410" alt="image" src="https://user-images.githubusercontent.com/50846373/236557382-2f715625-bbbf-451c-aa52-aeea6c18d9e5.png">

Reply:
Similar behavior in Windows! 

<img width="1330" alt="image" src="https://user-images.githubusercontent.com/50846373/236564076-50e94a92-d596-473a-ac5f-2bd74ce4d3f2.png">
<img width="1329" alt="Screenshot 2023-05-05 at 3 27 04 PM" src="https://user-images.githubusercontent.com/50846373/236564091-32c2799b-b37c-432b-8d48-937042c57f5f.png">